src/SelfServe/SelfServe.tsx

```diff
@@ -14,14 +16,35 @@ import { SelfServeDescriptor } from "./SelfServeTypes";
 import { SelfServeType } from "./SelfServeUtils";
 initializeIcons();
 
+const loadTranslationFile = async (className: string): Promise<void> => {
+  const language = i18n.languages[0];
+  const fileName = `${className}.json`;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let translations: any;
+  try {
+    translations = await import(`../Localization/${language}/${fileName}`);
+  } catch (e) {
+    translations = await import(`../Localization/en/${fileName}`);
+  }
+  i18n.addResourceBundle(language, className, translations.default, true);
+};
+
+const loadTranslations = async (className: string): Promise<void> => {
+  await loadTranslationFile("Common");
+  await loadTranslationFile(className);
+};
+
 const getDescriptor = async (selfServeType: SelfServeType): Promise<SelfServeDescriptor> => {
   switch (selfServeType) {
     case SelfServeType.example: {
       const SelfServeExample = await import(/* webpackChunkName: "SelfServeExample" */ "./Example/SelfServeExample");
+      await loadTranslations(SelfServeExample.default.name);
       return new SelfServeExample.default().toSelfServeDescriptor();
     }
     case SelfServeType.sqlx: {
       const SqlX = await import(/* webpackChunkName: "SqlX" */ "./SqlX/SqlX");
+      await loadTranslations(SqlX.default.name);
```

Comment:
Is it possible (and would it make sense) to infer the name from the class and load this for all the cases? ie moving `loadTranslations()` out of the switch so we only have one call.

Reply:
To me , it seems like more lines of code. We need to keep track of both the name and descriptor in each case and finally call loadTranslations() and return the descriptor.